src/main/java/com/google/sps/servlets/ColumnsFromTablesServlet.java

```diff
@@ -48,35 +47,35 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
       String[] listOfTables = request.getParameterValues(constant.TABLE_SELECT_PARAM);
       String database = request.getParameter(constant.DATABASE_PARAM);
 
+      this.dbClient = DatabaseConnector.getInstance().getDbClient(database);
 
-      Spanner spanner = SpannerOptions.newBuilder().build().getService();
-      DatabaseId db = DatabaseId.of("play-user-data-beetle", "test-instance", database);
-      this.dbClient = spanner.getDatabaseClient(db);
+      String query = "";
+      String selectedTables = "";
 
-      if(listOfTables.length == 1){
-          constant.QUERY = constant.QUERY + constant.GET_COLUMNS_FROM_TABLES + "\'" + listOfTables[0] + "\'" + constant.GROUP_BY_TABLE_NAMES;
-      }else{
-          constant.QUERY = constant.QUERY + constant.GET_COLUMNS_FROM_TABLES;
-          for(int i = 0; i < listOfTables.length; i++){
-              if(i != listOfTables.length-1){
-                constant.SELECTED_TABLES = "\'" + listOfTables[i] + "\'";
-                constant.QUERY = constant.QUERY + constant.SELECTED_TABLES + constant.COMMA;
-              }else{
-                  constant.SELECTED_TABLES = "\'" + listOfTables[i] + "\'";
-                  constant.QUERY = constant.QUERY + constant.SELECTED_TABLES;
-                }
-            }
-          constant.QUERY = constant.QUERY + constant.GROUP_BY_TABLE_NAMES;   
+      if (listOfTables.length == 1) {
+        query = query + constant.GET_COLUMNS_FROM_TABLES + "\'" + listOfTables[0] + "\'" + constant.GROUP_BY_TABLE_NAMES;
+      } else {
+        query = query + constant.GET_COLUMNS_FROM_TABLES;
+        for (int i = 0; i < listOfTables.length; i++) {
+          if (i != listOfTables.length-1) {
+            selectedTables = "\'" + listOfTables[i] + "\'";
```

Comment:
Have you tried just put tables in quotes without backslash? I was trying to find cloud spanner java documents about taking the raw string, but didn't find any good one, reading from this example, it seems we can just pass in single quotes without the escape backslash. https://cloud.google.com/spanner/docs/getting-started/java#write-data.

Reply:
I think this is mainly Millennia's servlet, so I'll pass this comment on to her and discuss together.

Reply:
@hilakey